rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/RestControllerHelper.java

```diff
@@ -243,7 +242,30 @@ private void addEmbeddedAttachments(
                 log.error("cannot create embedded attachment with content id: " + attachment.getAttachmentContentId());
             }
 
-            halResource.addEmbeddedResource("attachments", halAttachmentResource);
+            halResource.addEmbeddedResource("sw360:attachments", halAttachmentResource);
+        }
+    }
+
+    public void addEmbeddedProject(HalResource<Project> halProject, Set<String> projectIds, Sw360ProjectService sw360ProjectService, User user) throws TException {
+        for (String projectId : projectIds) {
+            final Project project = sw360ProjectService.getProjectForUserById(projectId, user);
+            addEmbeddedProject(halProject, project);
+        }
+    }
+
+    private void addEmbeddedProject(HalResource halResource, Project project) {
+        Project minimalFilledProject = new Project(project.getName());
+        minimalFilledProject.setType(null);
+        minimalFilledProject.setProjectType(null);
+        HalResource<Project> halProject = new HalResource<>(minimalFilledProject);
+
+        try {
+            Link projectLink = linkTo(ProjectController.class).slash("api" + ProjectController.PROJECTS_URL + "/" + project.getId()).withSelfRel();
+            halProject.add(projectLink);
+        } catch (Exception e) {
```

Comment:
why the generic `catch`? can't it be removed?

Reply:
its not necessary here... copied code. corrected all occurrences